Tests/GRPCTests/ServerErrorDelegateTests.swift

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+import XCTest
+import NIO
+import NIOHTTP1
+import GRPC
+import EchoModel
+import EchoImplementation
+import Logging
+
+private class ServerErrorDelegateMock: ServerErrorDelegate {
+  var transformLibraryErrorCallCount: Int = 0
+  var transformLibraryErrorHandler: ((Error) -> (GRPCStatusAndMetadata?))?
+  func transformLibraryError(_ error: Error) -> GRPCStatusAndMetadata? {
+    transformLibraryErrorCallCount += 1
+    if let transformLibraryErrorHandler = transformLibraryErrorHandler {
+        return transformLibraryErrorHandler(error)
+    }
+    fatalError("transformLibraryError returns can't have a default value thus its handler must be set")
+  }
+}
+
+class ServerErrorDelegateTests: GRPCTestCase {
+  private var channel: EmbeddedChannel!
+  private var errorDelegate: ServerErrorDelegateMock!
+
+  override func setUp() {
+    super.setUp()
+
+    let provider = EchoProvider()
+    errorDelegate = ServerErrorDelegateMock()
+    let handler = GRPCServerRequestRoutingHandler(
+      servicesByName: [provider.serviceName: provider],
+      encoding: .disabled,
+      errorDelegate: errorDelegate,
+      logger: self.logger
+    )
+
+    self.channel = EmbeddedChannel(handler: handler)
+  }
+
+  override func tearDown() {
+    XCTAssertNoThrow(try self.channel.finish())
+  }
+
+  func testTransformLibraryError_whenTransformingErrorToStatus() throws {
+    errorDelegate.transformLibraryErrorHandler = { _ in
+      GRPCStatusAndMetadata(status: .processingError)
```

Comment:
`.processingError` is a bad choice here since it's the "default" error we fall back to. It'd be better to provide something more specific (with a status message as well).